### PR TITLE
[SPARK-38426] Fixes perms for actions

### PR DESCRIPTION
**Pull request by naveensrinivasan**
### What changes were proposed in this pull request?
- Fixed permission for some of workflow actions https://github.com/ossf/scorecard/blob/main/docs/checks.md#token-permissions
- Pinned some of the actions SHA https://github.com/ossf/scorecard/blob/main/docs/checks.md#pinned-dependencies

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
N/A

### CODE REVIEW

File: .github/workflows/benchmark.yml [hunk 1]
```diff
@@ -71,12 +76,12 @@ jobs:
       SPARK_HOME: ${{ github.workspace }}
     steps:
     - name: Checkout Spark repository
-      uses: actions/checkout@v2
+      uses: actions/checkout@629c2de402a417ea7690ca6ce3f33229e27606a5 # v2
```
Comment (HyukjinKwon) [inline .github/workflows/benchmark.yml:79]: Hm, I think we intentionally didn't pin this version because `actions` are considered as a safe one.

Comment (naveensrinivasan) [reply .github/workflows/benchmark.yml:79]: Ok. Is there a specific reason why you think pinning a version would be an issue?

Comment (HyukjinKwon) [reply .github/workflows/benchmark.yml:79]: I don't see a reason why we should necessarily pin the version since it will automatically leverage bug fixes.

Comment (naveensrinivasan) [reply .github/workflows/benchmark.yml:79]: Ok. If you want me to revert I can do that. Let me know.

Comment (HyukjinKwon) [reply .github/workflows/benchmark.yml:79]: Yeah, let's revert. This was actually already discussed when we pin other plugins IIRC.

Comment (naveensrinivasan) [reply .github/workflows/benchmark.yml:79]: I have reverted the changes.

**Comment by naveensrinivasan** [comment]
A friendly reminder

**Comment by naveensrinivasan** [comment]
@HyukjinKwon A friendly ping

**Comment by HyukjinKwon** [comment]
I am fine. Just don't feel strong enough to approve and merge. We can wait for other committers to join/merge/approve.

**Comment by naveensrinivasan** [comment]
> I am fine. Just don't feel strong enough to approve and merge. We can wait for other committers to join/merge/approve.

Ok. Is there a specific reason for your feeling?

**Comment by naveensrinivasan** [comment]
@HyukjinKwon A friendly ping

**Comment by srowen** [comment]
Seems like the same comment would go for not pinning other actions?
Does restricting to read permission have any negative effects at all for our usage? can't think of any

**Pull request closed without merging**